### ISSUE TITLE
Error on `Collection.write()` since schema updates in 1.7.4

### ISSUE THREAD

**Issue by SimonSimCity** [VeliovGroup/Meteor-Files#579]
Since I updated to `ostrio:files@1.9.6` I see myself facing an exception when calling `collection.write()` telling me that the `Mime is required in files insert`.

This is reproducible and mostly strikes me in my tests, where I use the following lines:

```js
import { FilesCollection } from 'meteor/ostrio:files';

const MyFiles = new FilesCollection({ collectionName: 'my' });
MyFiles.collection.attachSchema(MyFiles.schema);

MyFiles.write(Buffer.from('123', 'utf8'), {
  meta: {
    createdAt: new Date(),
  },
  fileName: 'backup3.zip',
  type: 'plain/text',
}, (writeError3) => {
  if (writeError3) {
    console.error(writeError3);
  } else {
    console.log('done');
  }
});
```

The reason for this error is, that you've changed the expected schema in version `1.9.6` to include fields like `mime` and `mime-type` - both of them as required fields.

The data, you are saving in https://github.com/VeliovGroup/Meteor-Files/blob/master/server.js#L1082 is provided by `this._dataToSchema` as you can see [here](https://github.com/VeliovGroup/Meteor-Files/blob/master/server.js#L1176). The function `_dataToSchema `, [described in core.js](https://github.com/VeliovGroup/Meteor-Files/blob/master/core.js#L165) returns an object which does neither include the first nor the latter parameter, which get's it to fail when calling `collection.insert()` because it doesn't match the newly defined schema.

My suggestion now is to either make them both optional (as it has been until now - but would be inconsistent) or update the `_dataToSchema()` method to add the two parameters to it's objects.

**Comment by dr-dimitru** [comment]
Hello @SimonSimCity ,

 1. Thank you for reporting about this issue;
 2. Please update this thread following our [issue template](https://github.com/VeliovGroup/Meteor-Files/blob/master/.github/ISSUE_TEMPLATE):
    - Version of `Meteor` you're experiencing this issue
    - Where this issue appears? OS (Mac/Win/Linux)? Browser name and its version?
    - Post *Client* and/or *Server* logs with enabled `debug` option, you can enable "debug" mode in [*Constructor*](https://github.com/VeliovGroup/Meteor-Files/wiki/Constructor)


@coagmano could you please step-in, as you were the last to contribute to the Schema validation process.

**Comment by coagmano** [comment]
I think that both the `_dataToSchema` method definitely should return an object that matches the schema, and that the legacy items in the schema like `mime` and `mime-type` should be optional.

I'll whip up a PR to do that now

**Comment by dr-dimitru** [comment]
@coagmano sounds good, waiting for PR.
Thank you for quick response on that one.

**Comment by dr-dimitru** [comment]
@coagmano @SimonSimCity this should be fixed in [`v1.9.8`](https://github.com/VeliovGroup/Meteor-Files/releases/tag/1.9.8), please upgrade to the latest release, and let me know how it's working on your end.

Feel free to reopen it in case if the issue is still persists on your end.
